src/paramiko_mock/sftp_mock.py

```diff
@@ -13,123 +15,146 @@ class SFTPFileSystem():
     """
     file_system: dict[str, "SFTPFileMock"] = {}
 
-    def add_file(self, path, content):
+    def add_file(self, path: str, content: "SFTPFileMock") -> None:
         self.file_system[path] = content
 
-    def get_file(self, path):
+    def get_file(self, path: str) -> Union["SFTPFileMock", None]:
         return self.file_system.get(path)
-    
-    def remove_file(self, path):
+
+    def remove_file(self, path: str) -> None:
         self.file_system.pop(path, None)
 
-    def list_files(self):
-        return self.file_system.keys()
+    def list_files(self) -> list[str]:
+        return list(self.file_system.keys())
+
 
 class SFTPFileMock():
     """
     This class mocks a file in the remote filesystem.
     """
-    write_history = []
-    file_content = None
+    write_history: list[Any] = []
+    file_content: Any = None
 
-    def close(self):
+    def close(self) -> None:
         pass
 
-    def write(self, data):
+    def write(self, data: Any) -> None:
         self.write_history.append(data)
         self.file_content = data
-    
-    def read(self, size=None):
+
+    def read(self, size: Union[int, None] = None) -> Any:
```

Comment:
Python 3.9 EOL was 2025-10-31, and 3.10+ natively support using |
```suggestion
    def read(self, size: int | None = None) -> Any:
```